src/Smoke.cpp

```diff
@@ -366,185 +378,184 @@ struct SmokeWidget : ModuleWidget {
-  
-  menu->addChild(construct<MenuItem>(&MenuItem::text, "HIFI/LOFI"));
-  menu->addChild(construct<CloudsLofiItem>(&MenuItem::text, "HIFI", &CloudsLofiItem::clouds, clouds, &CloudsLofiItem::setting, false));
-  menu->addChild(construct<CloudsLofiItem>(&MenuItem::text, "LOFI", &CloudsLofiItem::clouds, clouds, &CloudsLofiItem::setting, true));  
-  
+    menu->addChild(construct<MenuItem>(&MenuItem::text, "STEREO/MONO"));
+    menu->addChild(construct<CloudsMonoItem>(&MenuItem::text, "STEREO", &CloudsMonoItem::clouds, clouds, &CloudsMonoItem::setting, false));
+    menu->addChild(construct<CloudsMonoItem>(&MenuItem::text, "MONO", &CloudsMonoItem::clouds, clouds, &CloudsMonoItem::setting, true));  
+    
+    menu->addChild(construct<MenuItem>(&MenuItem::text, "HIFI/LOFI"));
+    menu->addChild(construct<CloudsLofiItem>(&MenuItem::text, "HIFI", &CloudsLofiItem::clouds, clouds, &CloudsLofiItem::setting, false));
+    menu->addChild(construct<CloudsLofiItem>(&MenuItem::text, "LOFI", &CloudsLofiItem::clouds, clouds, &CloudsLofiItem::setting, true));  
+    
 #ifdef BUFFERRESIZING
-// disable by default as it seems to make alternative modes unstable
-  menu->addChild(construct<MenuItem>(&MenuItem::text, "BUFFER SIZE (EXPERIMENTAL)"));
-  menu->addChild(construct<CloudsBufferItem>(&MenuItem::text, "ORIGINAL", &CloudsBufferItem::clouds, clouds, &CloudsBufferItem::setting, 1));
-  menu->addChild(construct<CloudsBufferItem>(&MenuItem::text, "2X", &CloudsBufferItem::clouds, clouds, &CloudsBufferItem::setting, 2));
-  menu->addChild(construct<CloudsBufferItem>(&MenuItem::text, "4X", &CloudsBufferItem::clouds, clouds, &CloudsBufferItem::setting, 4));
-  menu->addChild(construct<CloudsBufferItem>(&MenuItem::text, "8X", &CloudsBufferItem::clouds, clouds, &CloudsBufferItem::setting, 8));
+  // disable by default as it seems to make alternative modes unstable
+    menu->addChild(construct<MenuItem>(&MenuItem::text, "BUFFER SIZE (EXPERIMENTAL)"));
+    menu->addChild(construct<CloudsBufferItem>(&MenuItem::text, "ORIGINAL", &CloudsBufferItem::clouds, clouds, &CloudsBufferItem::setting, 1));
+    menu->addChild(construct<CloudsBufferItem>(&MenuItem::text, "2X", &CloudsBufferItem::clouds, clouds, &CloudsBufferItem::setting, 2));
+    menu->addChild(construct<CloudsBufferItem>(&MenuItem::text, "4X", &CloudsBufferItem::clouds, clouds, &CloudsBufferItem::setting, 4));
+    menu->addChild(construct<CloudsBufferItem>(&MenuItem::text, "8X", &CloudsBufferItem::clouds, clouds, &CloudsBufferItem::setting, 8));
 #endif  
-    
-  return menu;
-}
 
-Model *modelSmoke 	= Model::create<Smoke,SmokeWidget>( 	 "Southpole", "Smoke", 		"Smoke - texture synth", GRANULAR_TAG, REVERB_TAG);
+  }
+
+  void step() override {
+    Smoke *smoke = dynamic_cast<Smoke*>(module);
+
+    if (smoke) {
+      panel1->visible = true;
+      panel2->visible = false;
+      panel3->visible = false;
+      panel4->visible = false;
+#ifdef PARASITES
```

Comment:
A lot of the parasites stuff I generally left alone, it wasn't clear to me if that should be working or not (README seemed to imply that it was working on a branch build?)